src/Ocelot/Cache/OutputCacheMiddleware.cs

```diff
@@ -60,8 +60,16 @@ public async Task Invoke(HttpContext httpContext)
         cached = await CreateCachedResponse(downstreamResponse);
 
         var ttl = TimeSpan.FromSeconds(options.TtlSeconds);
-        _outputCache.Add(downStreamRequestCacheKey, cached, options.Region, ttl);
-        Logger.LogDebug(() => $"Finished response added to cache for the '{downstreamUrlKey}' key.");
+        if (options.StatusCodeFilter == null || options.StatusCodeFilter.PassesFilter(cached.StatusCode))
+        {
+            _outputCache.Add(downStreamRequestCacheKey, cached, options.Region, ttl);
+            Logger.LogDebug(() => $"Finished response added to cache for the '{downstreamUrlKey}' key.");
```

Comment:
Please wrap the logging for the Debug scenario with a preprocessor directive, as another team member will likely request this 👇 
```suggestion
#if DEBUG
            Logger.LogDebug(() => $"Finished response added to cache for the '{downstreamUrlKey}' key.");
#endif
```